koboldcpp.py

```diff
@@ -807,6 +809,33 @@ def getfilename(var, text):
         button = ctk.CTkButton(parent, 50, text="Browse", command= lambda a=var,b=searchtext:getfilename(a,b))
         button.grid(row=row+1, column=1, stick="nw")
         return
+    
+    from subprocess import run, CalledProcessError
```

Comment:
I don't think I want to include this part - clinfo is an external dependency and doesn't exist by default on most systems, and also starting arbitrary processes is something i'd like to avoid.

Reply:
That's understandable, but I know the whole function works with AMD GPUs on Windows and Linux . Doesn't Clinfo get automatically installed when OpenCL is installed on a PC? If not, are there any other ways of calling the GPU name?

Reply:
Hm, it actually does not exist on my device, even though I have OpenCL working fine. clinfo is not installed by default and must be built/obtained separately. (Also I obviously don't have rocminfo, but the way you currently implemented it will attempt to launch **any** executable in the system path named `clinfo` or `rocminfo` which is not a good idea either. And on devices with no GPU, trying to running a non-existent file and catching the exception for it feels wrong too.

Additionally, starting a new process to parse part of the text output is kind of a hacky way of doing it. What if the console output format changes from different clinfo/rocminfo versions? A proper implementation should be done via an APIs such as `clGetDeviceInfo` rather than starting a new process. From python I don't think there is a reliable cross platform way to do it before the kcpp dll has been loaded.

I think maybe lets get the rocm build functional and merge that first, and then you can look into getting the GPU identifiers separately, or optionally as a bonus feature within your fork.

Reply:
@YellowRoseCx the clinfo issue can be handled on Windows if we ship a recompiled clinfo.exe